src/viam/rpc/server.py

```diff
@@ -114,3 +131,21 @@ async def create_and_serve(
         """
         server = cls(components)
         await server.serve(host, port, log_level, path=path)
+
+
+def wrapper(func: Callable):
+    """
+    Wrap a function so that any exceptions get raised as GRPCErrors to the client.
+
+    Args:
+        func (Callable): The function that should be wrapped
+    """
+
+    async def function(*args, **kwargs):
+        try:
+            new_func = await func(*args, **kwargs)
+            return new_func
+        except Exception as e:
```

Comment:
You should add an additional except right above this to catch and re-raise grpc errors (so we don't wrap those as status unknown)

```python3
try:
    ...
except GRPCError:
    raise
except Exception as e:
    raise GRPCError(Status.UNKNOWN, e)
```